clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1073,22 +1084,14 @@ private void doAutoCommitOffsetsAsync() {
         });
     }
 
-    private void maybeAutoCommitOffsetsSync(Timer timer) {
+    private RequestFuture<Void> maybeAutoCommitOffsetsAsync() {
```

Comment:
I think these two overloaded `maybeAutoCommitOffsetsAsync` functions can be merged: compared with the other that returns `void`, this function did the following more:

1) `client.pollNoWakeup();` -> this has already been called inside the `commitOffsetsAsync` callee at https://github.com/apache/kafka/pull/11340/files#diff-0029e982555d1fae10943b862924da962ca8e247a3070cded92c5f5a5960244fR982, so this is not needed.

2) `invokeCompletedOffsetCommitCallbacks();` -> I think this is not required in the async call.

As a result, we can just leave one `maybeAutoCommitOffsetsAsync` function that returns the future.

Reply:
I agree with you,thanks